### ISSUE TITLE
Widescreen - max width for a window or make them only half of the screen

### ISSUE THREAD

**Issue by prgres** [koekeishiya/yabai#1499]
As I use a 38-inch widescreen monitor it is not quite comfortable to have the window stretch to full size, it is too wide and hardly maintain. I mean really. I have to open a placeholder chrome window along with my terminal to not get dizzy. It is pain in the ass to do this every time. 
I thought “hey, maybe set max width for each window for a half screen, and that would be okay". But I have not found any option for that (except using grid and floating windows but it breaks my borders and padding). 

To sum up, I would want an option to set the max width of the window on a particular monitor or add a layout option for that. Is there any possibility to add these and help me out?

**Comment by ephemeraleternity** [comment]
Since I've run into the same issue, I've written a script that uses yabais padding option to decrease window size when there are few windows on a desktop. This is quite hacky to be honest and requires a few workarounds to not be annoying, but it _does_ work.

```
 #!/bin/bash
currentsp=$(yabai -m query --spaces --space | jq '.index')
windowcount=$(yabai -m query --windows --space | jq -r '.[]."is-floating"' | grep false | wc -l)
excludes="Archive Utility\|Menuwhere\|Color Picker\|xQuartz\|VirtualBox WM"
countexcludes=$(yabai -m query --windows --space | jq -r '.[].app' | grep "${excludes}" | wc -l)
fscreenapps="Adobe Photoshop 2021\|Dreamweaver\|Adobe Illustrator 2021"
fscreen=$(yabai -m query --windows --space | jq -r '.[].app' | grep "${fscreenapps}" | wc -l)
hiddenwindows=$(yabai -m query --windows --space | jq -r '.[]."is-visible"' | grep false | wc -l)
windowcount="$((windowcount - hiddenwindows - countexcludes))"

if [[ "$currentsp" == 5 ]]; then
		yabai -m space --padding abs:100:100:800:800
	else
		if [ "$fscreen" -gt "0" ]; then
			yabai -m space --padding abs:20:20:20:20
		else
			case $windowcount in
				[0-1])
					yabai -m space --padding abs:100:100:950:950
					;;
				2)
					yabai -m space --padding abs:100:100:500:500
					;;
				3)
					yabai -m space --padding abs:70:70:70:70
					;;
				*)
					yabai -m space --padding abs:20:20:20:20
					;;
			esac
		fi
fi
```

You might want to adjust the padding values and probably remove the if function that checks whether space five is active. For some reason, adding "Archive Utility" to the Apps excluded by the script unfortunately doesn't work, so windows flash around for a second when you're extracting an archive. Counting hidden windows is also somewhat unreliable. Despite all of this, the script is pretty useable!

**Comment by prgres** [comment]
@ephemeraleternity Thanks! You inspired me and I have created something similar

`yabairc`
```
yabai -m signal 	--add event=window_destroyed 	action="$HOME/.config/yabai/padding.sh"
yabai -m signal 	--add event=window_focused 		action="$HOME/.config/yabai/padding.sh"
yabai -m signal 	--add event=window_resized 		action="$HOME/.config/yabai/padding.sh"
yabai -m signal 	--add event=window_moved 		action="$HOME/.config/yabai/padding.sh"
```

`padding.sh`
```
 #!/bin/bash

windowcount=$(yabai -m query --windows --space | jq -r '[ .[] | select(."is-floating" == false and ."subrole" != "AXDialog") ] | length' )
case "${windowcount}" in
        "1")
                display=$( yabai -m query --displays --space | jq )
                width=$( echo $display | jq '.frame.w')
                if [[ "${width}" != "1728" ]]; then
                        yabai -m space --padding abs:10:20:50:1500
                else
                        yabai -m space --padding abs:10:20:50:50
                fi
        ;;
        *)
                yabai -m space --padding abs:10:20:50:50
        ;;
esac
```

**Comment by prgres** [comment]
Okay, now I have another problem. After opening a new window, yabai needs some time to stabilize (?) everything which is pretty annoying

https://user-images.githubusercontent.com/23465248/198552006-1da77a17-ef6d-4bf0-a4fb-2b72110fd9e2.mp4

**Comment by ftqo** [comment]
I'm not sure if this should have been marked as completed. Would it be possible to add a configuration option to specify the maximum width a window can take up? I know something that looks similar can be done with the grid layout, is that the expected solution?
